toml/decoder.py

```diff
@@ -177,11 +192,28 @@ def loads(s, _dict=dict, decoder=None):
     keygroup = False
     dottedkey = False
     keyname = 0
+
+    arr = ''
+    comm_arr = False
+    arr_beg = arr_end = None
+    comm_list = []
```

Comment:
Can the `comm_list` go directly inside the decoder? This avoids the save step at the end of the first pass, and helps move a bit more logic inside the decoder.